pyproject.toml

```diff
@@ -10,6 +10,7 @@ dynamic = ["version"]
 dependencies = [
     "razorpay~=1.4.2",
     "PyGithub~=2.3.0",
+    "ics~=0.7.2",
```

Comment:
[icalendar](https://pypi.org/project/icalendar/) seems a good alternative ?

Reply:
I can try replacing this with icalender if that's what we want?

Reply:
I would suggest a better alternative would be to not take it to the backend at all. 

There is a time-factor involved. When I click the button, there is atleast a 2-3 s delay without any loading state in the UI - a bit confusing at first. 

We could potentially handles this on the client side - there should be JS libraries that supports calendaring. The entire data needed is there in the frontend. Handling the generation of an ics on the client side would make it faster too.

Reply:
This might be a helpful JS library here to use - [ics](https://www.npmjs.com/package/ics).

Reply:
I genuinely don't feel like this should be something we should throw away at client devices to handle 😅 

Moreover, I feel data can be fetched more easily (event title, event date, event location) in the server side using Frappe apis, and easily added to ics files

Reply:
> I genuinely don't feel like this should be something we should throw away at client devices to handle 😅

Can you please explain why? 

IMO the data is literally there in the frontend - why add an additional server hop to this ? Event date. Event title are also there - for the header ? Event location can also be something that's fetched alongside these event details.

Reply:
> Can you please explain why?

Not really, I just find that method to be something which i find to be quicker and easier to implement. I could be wrong too. Please start a new issue/discussion to discuss what method/technologies would be better to use in this regard. Not addressing this in this PR

Reply:
@harshtandiya, I do agree with @RiyaMathew-11  here about doing this on the client side instead of the server side.

The only reason why we're being forced to make an API call is because `FOSSEventSchedule` does not contain the `event_location` and we should just update the doctype to include the `event_location` attribute, set to the event location from the `FOSSChapterEvent` doctype

Once we do that, the client has all of the information in the session object and it doesn't need to make an API call at all, at which point we can just use the `ics` npm package to create the ICS file.

Along with Riyas' argument about better performance because of one less API call, I advocate for doing this on the client side because it reduces our API surface area. It's a low risk API but it's still one more API endpoint and anytime we can get away with adding less code, we should IMHO.

@harshtandiya can we please make this change in this PR? On the bright side, you can ignore my comment about the unittest because the endpoint itself won't exist :D